### DISCUSSION TITLE
getRowHeight in aggrid

### DISCUSSION THREAD

**Discussion by aersam** [zauberzeug/nicegui#1468] [Ideas / Feature Requests]
Hi there!

I would like to be able to use the [getRowHeight Callback](https://www.ag-grid.com/javascript-data-grid/row-height/#getrowheight-callback) in aggrid. In order to implement this I thought about a Python Class wrapping a JavaScript [Function constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/Function). The server would pass the Python object and in JS it would be turned into a Function constructor. 

Python:
```python
ui.aggrid(
{                
       "columnDefs": [
             {
                "getRowHeight": JSFunction("params", "return params.data.myCol == 1 ? 50: 20"),
...
}])

```
JS:
```javascript
      for (const col in Object.values(this.gridOptions.columnDefs)) {
        col["getRowHeight"] = get_js_func(col["getRowHeight"])
      }

...
}
```

Would also do a PR :) I think the JSFunction concept would be meaningful in other places, too

**Comment by falkoschindler** [comment]
Thanks for the suggestion, @aersam!
I wonder, what would be the advantage over the approach we used for `ui.table` (e.g. in the [sorting and formatting demo](https://nicegui.io/documentation/table#custom_sorting_and_formatting))?
```py
columns = [
    {
        ...
        ':sort': '(a, b, rowA, rowB) => b.length - a.length',
    },
    {
        ...
        ':format': 'value => value + " years"',
    },
]
```

**Reply by aersam** [reply]
Ah yes, no advantage, sorry! I just did not know about this feature, sorry! thanks a lot!

**Reply by aersam** [reply]
Would that work already with getRowHeight in aggrid?

**Reply by falkoschindler** [reply]
No, it's not yet implemented. Here is the corresponding feature request: https://github.com/zauberzeug/nicegui/discussions/1420.

**Reply by aersam** [reply]
Ok, here's the pull request for it: https://github.com/zauberzeug/nicegui/pull/1470 :)

**Comment by falkoschindler** [comment]
Dynamic properties for `ui.aggrid` have been released in version 1.3.12.
A demo using `getRowHeight` will be deployt to the documentation page shortly.

**Reply by rodja** [reply]
It's now live at https://nicegui.io/documentation/aggrid#ag_grid_with_dynamic_row_height
